frontend/test/metabase/scenarios/models/model-actions.cy.spec.js

```diff
@@ -0,0 +1,201 @@
+        id: TEST_PARAMETER.id,
+        required: true,
+        fieldType: "number",
+        inputType: "number",
+      },
+    },
+  },
+};
+
+describe("scenarios > models > actions", () => {
+  beforeEach(() => {
+    restore("postgres-12");
+    cy.signInAsAdmin();
+
+    enableActionsForDB();
+    enableActionsForDB(PG_DB_ID);
+
+    cy.createQuestion(SAMPLE_ORDERS_MODEL, {
+      wrapId: true,
+      idAlias: "modelId",
+    });
+
+    cy.intercept("/api/card/*").as("getModel");
+  });
+
+  it("should allow to view, create and edit model actions", () => {
+    cy.get("@modelId").then(id => {
+      cy.visit(`/model/${id}/detail`);
+      cy.wait("@getModel");
+    });
+
+    cy.findByText("Actions").click();
+
+    cy.findByRole("button", { name: /Create basic actions/i }).click();
+    cy.findByLabelText("Action list").within(() => {
+      cy.findByText("Create").should("be.visible");
+      cy.findByText("Update").should("be.visible");
+      cy.findByText("Delete").should("be.visible");
+    });
+
+    cy.findByRole("button", { name: "New action" }).click();
+    fillQuery("DELETE FROM orders WHERE id = {{ id }}");
+    fieldSettings()
+      .findByText(/Number/i)
+      .click();
+    cy.findByRole("button", { name: "Save" }).click();
+    modal().within(() => {
+      cy.findByLabelText("Name").type("Delete Order");
+      cy.findByRole("button", { name: "Create" }).click();
+    });
+    cy.findByLabelText("Action list")
+      .findByText("Delete Order")
+      .should("be.visible");
+
+    openActionEditorFor("Delete Order");
+    fillQuery(" AND status = 'pending'");
+    fieldSettings()
+      .findByRole("radiogroup", { name: /Field type/i })
+      .findByLabelText(/Number/i)
+      .should("be.checked");
+    cy.findByRole("button", { name: "Update" }).click();
+
+    cy.findByLabelText("Action list")
+      .findByText(
+        "DELETE FROM orders WHERE id = {{ id }} AND status = 'pending'",
+      )
+      .should("be.visible");
+  });
+
+  it("should allow to make actions public and execute them", () => {
+    cy.intercept("/api/public/action/*/execute", request => {
+      expect(request.body).to.deep.equal({
+        parameters: { [TEST_PARAMETER.id]: -2 },
+      });
+    });
+
+    cy.get("@modelId").then(modelId => {
+      cy.request("POST", "/api/action", {
+        ...SAMPLE_QUERY_ACTION,
+        model_id: modelId,
+      });
+      cy.visit(`/model/${modelId}/detail`);
+      cy.wait("@getModel");
+    });
+
+    cy.findByText("Actions").click();
+    openActionEditorFor(SAMPLE_QUERY_ACTION.name);
+
+    cy.findByRole("dialog").within(() => {
+      cy.findByRole("button", { name: "Action settings" }).click();
+      cy.findByLabelText("Make public").should("not.be.checked").click();
+      cy.findByLabelText("Public action link URL")
+        .invoke("val")
+        .then(url => {
+          cy.wrap(url).as("publicUrl");
+        });
+    });
+
+    cy.get("@publicUrl").then(url => {
+      cy.signOut();
```

Comment:
Very cool! This confirms that we don't accidentally use the admin credential to view the public action.